src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -412,6 +414,24 @@ static RegexCharClass()
             int len = s_propTable.Length;
             for (int i = 0; i < len - 1; i++)
                 Debug.Assert(string.Compare(s_propTable[i][0], s_propTable[i + 1][0], StringComparison.Ordinal) < 0, $"RegexCharClass s_propTable is out of order at ({s_propTable[i][0]}, {s_propTable[i + 1][0]})");
+
+            CultureInfo culture = CultureInfo.InvariantCulture;
+            for (int k = 0; k < s_lcTable.Length; k++)
+            {
+                LowerCaseMapping loc = s_lcTable[k];
+                if (loc.LcOp == 1)
+                {
+                    // Validate only the LowercaseAdd cases
```

Comment:
Why not validate all the table while you are at it?

Reply:
Good idea. Done.

Reply:
Hmm, as part of validating the table I ran into this case. Results from the immediate window below:

```
uppercase
304 'İ'
culture.TextInfo.ToLower(uppercase)
304 'İ'
char.ToLower(uppercase) // This is what is stored in s_lcTable
105 'i'
```

`culture` here (and in the validation code in this PR) is set to `CultureInfo.InvariantCulture`. `char.ToLower` uses `CurrentCulture` instead. This behavior is related to https://github.com/dotnet/runtime/issues/36147. Essentially, `s_lcTable` seems to have its values populated for "en-US".